### PR TITLE
Fix: Local adapter check address on stage

### LINKED ISSUE #5361: [Bug]: StageObject API for local block adapter doesn't check if local import is enabled
### What happened?

Current Behavior:
I was able to import a single object to lakeFS running on local blockstore, but not to read it.
        
Steps to Reproduce:
1. Stage a file:
`lakectl fs stage lakefs://repo1/main/1.txt --location local:///tmp/import/bla.txt --checksum 1 --size 10`
2. Try to download it (like thru the UI) and fail.

### Expected Behavior

Expected Behavior:
If lakeFS allows me to stage an object, then I should be able to download it.
We need to make sure that both `local.import_enabled` enabled and `local.allowd_external_prefixes` contains the source when staging a local object. Right now it's not checked when staging an object. It is checked when downloading the object. 

### lakeFS Version

0.94.1

### Deplyoment

local

### Affected Clients

All

### Relevant logs output

_No response_

### Contact Details

_No response_

### PR DESCRIPTION
**Pull request by N-o-Z**
### Linked Issue

Closes #5361

---

### Background

Enabling local import - we need to check that stageObject api doesn't allow staging physical addresses which are not allowed by the local block adapter

### Bug Fix

Added check when resolving namespace to return error if address is not part of the block adapter's allowed prefixes

### Testing Details

Added new controller unit test for this specific scenario

### Breaking Change?

Nyet

### CODE REVIEW

File: pkg/block/local/adapter.go [hunk 1]
```diff
@@ -542,7 +542,17 @@ func (l *Adapter) GetStorageNamespaceInfo() block.StorageNamespaceInfo {
 func (l *Adapter) ResolveNamespace(storageNamespace, key string, identifierType block.IdentifierType) (block.QualifiedKey, error) {
 	qk, err := block.DefaultResolveNamespace(storageNamespace, key, identifierType)
 	if err != nil {
-		return qk, err
+		return nil, err
+	}
+
+	// Check if path allowed and return error if path is not allowed
+	_, err = l.extractParamsFromObj(block.ObjectPointer{
```
Comment (itaiad200) [inline pkg/block/local/adapter.go:549]: Are we sure that's the appropriate location to check it? It seems weird to have a `ResolveNamespace` method checking permissions using `extract..` . Maybe we should use other names for the `extract` function?

**Review by itaiad200 — APPROVED** [review]
Approved with a single comment :)

File: pkg/block/local/adapter.go (hunk 1 above)
Comment (N-o-Z) [reply pkg/block/local/adapter.go:549]: I think I'll just *extract* the path validation from the extract method 😃

**Review by nopcoder — CHANGES_REQUESTED** [review]

File: pkg/block/local/adapter.go (hunk 1 above)
Comment (N-o-Z) [reply pkg/block/local/adapter.go:549]: Tried to separate the logic, but it doesn't feel natural. Prefer to leave it as is - I think this is the lesser evil 😄

File: pkg/api/serve_test.go [hunk 2]
```diff
@@ -121,7 +121,10 @@ func createDefaultAdminUser(t testing.TB, clt api.ClientWithResponsesInterface)
 func setupHandlerWithWalkerFactory(t testing.TB, factory catalog.WalkerFactory) (http.Handler, *dependencies) {
 	t.Helper()
 	ctx := context.Background()
-	viper.Set("blockstore.type", block.BlockstoreTypeMem)
+
+	if viper.Get(config.BlockstoreTypeKey) == nil {
+		viper.Set(config.BlockstoreTypeKey, block.BlockstoreTypeMem)
+	}
```
Comment (nopcoder) [inline pkg/api/serve_test.go:127]: The previous code assumed that the package level will use blockstore of mem type as global state - using `if` here insidate that we may change the state, which will make an issue when we will run some tests in parallel.
Will it be possble to setup the specific adater in a specific test without modify the viper global state?

**Review by nopcoder — APPROVED** [review]
Looks good. Suggest that ErrForbidden and ErrBadPath may include the path or prefix that caused the error - for the case we will have environemnt configured with set of paths trying to load metadata and fail. And we dont know the reason of the failure.

**Pull request merged**